databricks/sdk/mixins/files.py

```diff
@@ -467,6 +565,75 @@ def __repr__(self) -> str:
         return f'<_VolumesPath {self._path}>'
 
 
+class _ModelsPath(_Path):
```

Comment:
Do we need a new Path implementation? It seems like this just uses the Files API under the hood. From what I can tell, the only difference is `open()`, which returns ModelsIO now. That is in turn almost identical to _FilesIO.

If we changed https://github.com/databricks/databricks-sdk-py/pull/750/files#diff-469b96ba8e49ef7cbe8d3b4577d2a0ddf3a3552638b76d87bfccb936fda65d18R759 to check whether the path started with either `/Volumes` or `/Models`, I think the existing implementation would work. Is this correct? If so, we should just rename that to `_FilesPath` (as it uses the Files API) and reuse it.

Ultimately, that `_path` method should probably use some kind of prefix allowlist to determine whether the `_FilesPath` should be used, and we can add `/Volumes` and `/Models` as the initial prefixes.

Reply:
@mgyucht that sounds better. thanks for the suggestion, updated.

(I am not in the osp team but offered to do this work so we can speed up mlflow adoption of the databricks sdk, but I think since it's all Files API / Presigned URLs behind this sdk, we should be able to reuse the same path and io class for all securables)